src/conversion.rs

```diff
@@ -34,156 +37,286 @@ impl fmt::Display for Error {
-            .into()
-    });
+    fn from(polygon: &'a Polygon<T>) -> geo_types::Geometry<T> {
+        Self::Polygon(polygon.clone().into())
+    }
+}
 
-    match iter.next() {
-        Some(interior) => geo_types::Polygon::new(interior, iter.collect()),
-        None => geo_types::Polygon::new(geo_types::LineString(vec![]), vec![]),
+impl<T> From<Polygon<T>> for geo_types::Polygon<T>
+where
+    T: num_traits::Float,
+{
+    fn from(polygon: Polygon<T>) -> Self {
+        let mut iter = polygon.0.into_iter().map(geo_types::LineString::from);
+        match iter.next() {
+            Some(interior) => geo_types::Polygon::new(interior, iter.collect()),
+            None => geo_types::Polygon::new(geo_types::LineString(vec![]), vec![]),
+        }
     }
 }
 
-impl<'a, T> Into<geo_types::Geometry<T>> for &'a Polygon<T>
+impl<'a, T> TryFrom<&'a MultiPoint<T>> for geo_types::Geometry<T>
 where
     T: num_traits::Float,
 {
-    fn into(self) -> geo_types::Geometry<T> {
-        w_polygon_to_g_polygon(self).into()
+    type Error = Error;
+
+    fn try_from(multi_point: &'a MultiPoint<T>) -> Result<Self, Self::Error> {
+        Ok(Self::MultiPoint(multi_point.clone().try_into()?))
     }
 }
 
-impl<'a, T> Into<geo_types::Geometry<T>> for &'a MultiPoint<T>
+impl<T> TryFrom<MultiPoint<T>> for geo_types::MultiPoint<T>
 where
     T: num_traits::Float,
 {
-    fn into(self) -> geo_types::Geometry<T> {
-        let geo_multipoint: geo_types::MultiPoint<T> = self
+    type Error = Error;
+
+    fn try_from(multi_point: MultiPoint<T>) -> Result<Self, Self::Error> {
+        let points: Vec<geo_types::Point<T>> = multi_point
             .0
-            .iter()
-            .filter_map(|p| p.0.as_ref())
-            .map(|c| create_geo_coordinate(c))
-            .collect();
+            .into_iter()
+            .map(geo_types::Point::try_from)
+            .collect::<Result<Vec<_>, _>>()?;
 
-        geo_multipoint.into()
+        Ok(geo_types::MultiPoint(points))
     }
 }
 
-impl<'a, T> Into<geo_types::Geometry<T>> for &'a MultiPolygon<T>
+impl<'a, T> From<&'a MultiPolygon<T>> for geo_types::Geometry<T>
 where
     T: num_traits::Float,
 {
-    fn into(self) -> geo_types::Geometry<T> {
-        let geo_multipolygon: geo_types::MultiPolygon<T> =
-            self.0.iter().map(|p| w_polygon_to_g_polygon(p)).collect();
+    fn from(multi_polygon: &'a MultiPolygon<T>) -> Self {
+        Self::MultiPolygon(multi_polygon.clone().into())
+    }
+}
+
+impl<T> From<MultiPolygon<T>> for geo_types::MultiPolygon<T>
+where
+    T: num_traits::Float,
+{
+    fn from(multi_polygon: MultiPolygon<T>) -> Self {
+        let geo_polygons: Vec<geo_types::Polygon<T>> = multi_polygon
+            .0
+            .into_iter()
+            .map(geo_types::Polygon::from)
+            .collect();
 
-        geo_multipolygon.into()
+        geo_types::MultiPolygon(geo_polygons)
     }
 }
 
+#[deprecated(since = "0.8.1", note = "use `geometry_collection.try_into()` instead")]
```

Comment:
Since we're pre v1, I prefer to only bump the patch version, even when adding new functionality to avoid locking people to an unnecessarily old version. 

As a side note, I'd love to start v1'ing our crates.